website/source/docs/connect/proxies/envoy.md

```diff
@@ -0,0 +1,192 @@
+build envoy directly then the [`consul connect envoy`
+command](/docs/commands/connect/envoy.html) command can be used directly on your
+local machine to start envoy, however for this guide we'll use the Docker image.
+
+While the [`consul connect envoy` command](/docs/commands/connect/envoy.html)
+supports generating bootstrap config on the host that we could then mount in to
+the standard envoy Docker container, it's simpler to be able to use it to run
+docker directly which requires both consul and envoy binaries in one container.
+
+Using Docker 17.05 or higher (with multi-stage builds), create a Dockerfile with
+the following content:
+
+```text
+FROM consul:latest
+FROM envoyproxy/envoy:v1.8.0
+COPY --from=0 /bin/consul /bin/consul
+ENTRYPOINT ["dumb-init", "consul", "connect", "envoy"]
+```
+
+This takes the consul binary from the latest release image and copies it into a
+new image based on the official envoy build.
+
+This can be built locally with:
+
+```text
+docker build -t consul-envoy .
+```
+
+### Agent Setup
+
+To get a simple example working, run a single agent in `-dev` mode using the
+latest Docker image. It's possible to run the agent on the Docker host but it
+complicates the rest of the steps below due to networking config on different
+platforms. This guide will start all containers using Docker `host` network
+which is not a good simulation of a production setup, but makes the following
+steps much simpler.
+
+-> **Note:** `-dev` mode enables the gRPC server on port 8502 by default. For a
+production agent you'll need to [explicitly configure the gRPC
+port](/docs/agent/options.html#grpc_port).
+
+In order to start a proxy instance, a [proxy service
+definition](/docs/connect/proxies.html) must exist on your local agent. The
+simplest way to create one is using the [sidecar service
+registration](/docs/connect/proxies/sidecar-service.html) syntax.
+
+Create a config file called `envoy_demo.hcl` containing the following service
+definitions.
+
+```text
+services {
+  name = "web"
+  port = 8080
+  connect {
+    sidecar_service {
+      proxy {
+        upstreams {
+          destination_name = "db"
+          local_bind_port = 9191
+        }
+      }
+    }
+  }
+}
+services {
+  name = "db"
+  port = 9090
+  connect {
+    sidecar_service {}
+  }
+}
+```
+
+Consul agent can now be started in dev mode using host networking with that
+config:
+
+```text
+$ docker run --rm -d -v$(pwd)/envoy_demo.hcl:/etc/consul/envoy_demo.hcl \
+  --network host consul:latest \
+  agent -dev -config-file /etc/consul/envoy_demo.hcl
+```
+
+### Running Envoy
+
+With the above setup, envoy proxies for each service instance can now be run
+using:
+
+```text
+$ docker run --rm -d --network host \
+  consul-envoy -sidecar-for web
+3f213a3cf9b7583a194dd0507a31e0188a03fc1b6e165b7f9336b0b1bb2baccb
+$ docker run --rm -d --network host \
+  consul-envoy -sidecar-for db -admin-bind localhost:19001
```

Comment:
I would just make a note somewhere in the docs why this `-admin-bind` flag is necessary.